packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,267 @@
+  return addresses[0]
+}
+
+export const getConfirmationInput = (
+  transactionData: TransactionData,
+  e164NumberToAddress: E164NumberToAddressType,
+  manuallyValidatedE164NumberToAddress: ManuallyValidatedE164NumberToAddress
+): ConfirmationInput => {
+  const { recipient } = transactionData
+  let recipientAddress: string | null | undefined
+
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    recipientAddress = recipient.address
+  } else {
+    if (!recipient.e164PhoneNumber) {
+      throw new Error('Phone number missing')
+    }
+
+    recipientAddress = getAddressFromPhoneNumber(
+      recipient.e164PhoneNumber,
+      e164NumberToAddress,
+      manuallyValidatedE164NumberToAddress
+    )
+  }
+
+  return { ...transactionData, recipientAddress }
+}
+
+export const getVerificationStatus = (
+  recipient: Recipient,
+  e164NumberToAddress: E164NumberToAddressType
+): RecipientVerificationStatus => {
+  let recipientVerificationStatus
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    recipientVerificationStatus = RecipientVerificationStatus.VERIFIED
+  } else {
+    if (!recipient.e164PhoneNumber) {
+      throw new Error('Phone number missing')
+    }
+
+    const addresses = e164NumberToAddress[recipient.e164PhoneNumber]
+    if (addresses === undefined) {
+      recipientVerificationStatus = RecipientVerificationStatus.UNKNOWN
+    } else if (addresses === null) {
+      recipientVerificationStatus = RecipientVerificationStatus.UNVERIFIED
+    } else {
+      recipientVerificationStatus = RecipientVerificationStatus.VERIFIED
+    }
+  }
+
+  return recipientVerificationStatus
+}
+
+export const getFeeType = (
+  recipientVerificationStatus: RecipientVerificationStatus
+): FeeType | null => {
+  switch (recipientVerificationStatus) {
+    case RecipientVerificationStatus.UNKNOWN:
+      return null
+    case RecipientVerificationStatus.UNVERIFIED:
+      return FeeType.INVITE
+    case RecipientVerificationStatus.VERIFIED:
+      return FeeType.SEND
+  }
+}
+
+// Given addresses can be added and deleted we can't rely on changes in length to signal address changes
+// Not sure if address order im array is consistent so not assuming it
+const checkIfItemsOfChildArrAreSubsetOfParentArr = (parentArr: string[], childArr: string[]) => {
+  const parentArrSorted = parentArr.sort()
+  const childArrSorted = childArr.sort()
+
+  for (let i = 0; i < childArrSorted.length; i += 1) {
+    if (parentArrSorted[i] !== childArrSorted[i]) {
+      return false
+    }
+  }
+
+  return true
+}
+
+const checkIfNewAddressesAdded = (
+  oldAddresses: string[] | null | undefined,
+  newAddresses: string[]
+) => {
+  if (oldAddresses && checkIfItemsOfChildArrAreSubsetOfParentArr(oldAddresses, newAddresses)) {
+    return false
+  }
+
+  return true
+}
+
+const checkIfLast4DigitsAreUnique = (addressArr: string[]) => {
```

Comment:
Nit: I would find this helper functions easier to read if checkIf were removed from the beginning of the function names, but it's a matter of personal preference